main-archive.js

```diff
@@ -0,0 +1,175 @@
+      cell.style.color = "red";
+  }, 150);
+  setTimeout(function() {
+    cell.style.color = "";
+  }, 650);
+  setTimeout(function() {
+    cell.style.color = "red";
+  }, 1150);
+  setTimeout(function() {
+    cell.style.color = "";
+  }, 1650);
+}
+
+//Checking for a winner - Be more clever later
+function checkforWinner(){
+  let row1 = cell0.textContent+cell1.textContent+cell2.textContent; //These are all the winning combinations
+  let row2 = cell3.textContent+cell4.textContent+cell5.textContent;
+  let row3 = cell6.textContent+cell7.textContent+cell8.textContent;
+  let col1 = cell0.textContent+cell3.textContent+cell6.textContent;
+  let col2 = cell1.textContent+cell4.textContent+cell7.textContent;
+  let col3 = cell2.textContent+cell5.textContent+cell8.textContent;
+  let diagL = cell0.textContent+cell4.textContent+cell8.textContent;
+  let diagR = cell2.textContent+cell4.textContent+cell6.textContent;
+  let checkAnswer = [row1, row2, row3, col1, col2, col3, diagL, diagR]; //All winning combinations added to array for iteration
+
+  //checks to see if there's a winning or drawn state, if so it gives an alert and calls resetBoard to reset the game board 
+  checkAnswer.forEach((cond)=>{
+    if(cond === "XXX"){
+      //alert('You win!')
+      status.textContent = "You win!"
+      gameOver = true;
+      resetBoard()
+    }
+    if(cond === "OOO"){
+      alert('The computer wins!')
+      resetBoard()
+    }
+    //checks if the game has started (to stop drawing on an empty board when the game starts) and if all the cells have text content
+  })
+  if(gameStarted && allCells.every(element => element.textContent !== "")){
+    alert(`It's a draw!`);
+    resetBoard();
+  }
+}
+
+//Reset the board and re-enable the start button
+function resetBoard(){
+  allCells.forEach(cell => cell.textContent = "");
+  gameStarted = false;
+  startButton.disabled = false;
+  playerTurn = false;
+}
+
+//Timer
+function timerFunction(){
+  let seconds = 1; //Set to one when the timer starts because the next call isn't for another second
+  let mins = 0;
+  let hours = 0;
+  let gameTime = window.setInterval(countSecs, 1000); //This is getting restarted every time the game starts so it will count by 2-3 seconds
+  let timer = document.getElementById('timer');
+  function countSecs(){
+    //resetBoard() changes gameStarted to false when the game is won
+    if(!gameStarted){
+      clearInterval(gameTime);
+      seconds = 0;
+    }
+    if(seconds==60){ //Ticks up mins after 60 seconds
+      mins++;
+      seconds = 0;
+    }
+    if(mins==60){
+      hours++;
+      mins = 0;
+    } 
+    if(gameOver){
+      clearInterval(gameTime);
+    }
+    timer.textContent = `Time Elapsed: ${hours.toString().padStart(2,"0")}:${mins.toString().padStart(2,"0")}:${seconds.toString().padStart(2,"0")}`;
+    seconds++;
+  }
+}
+
+//Computer picks a cell at random from an array of empty cells
+let compButton = document.getElementById("computer");
```

Comment:
It's best practice to have all your global variables defined at the top of the function, to prevent accidentally re using names, and to make sure they're always defined before they're used.